rust/src/partitions.rs

```diff
@@ -0,0 +1,100 @@
+use std::convert::TryFrom;
+
+use crate::DeltaTableError;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum PartitionValue<T> {
+    Equal(T),
+    NotEqual(T),
+    In(Vec<T>),
+    NotIn(Vec<T>),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct PartitionFilter<'a, T> {
+    pub partition_key: &'a str,
+    pub partition_value: PartitionValue<T>,
+}
+
+impl<'a, T: std::fmt::Debug> TryFrom<(&'a str, &str, T)> for PartitionFilter<'a, T> {
+    type Error = DeltaTableError;
+
+    fn try_from(filter: (&'a str, &str, T)) -> Result<Self, DeltaTableError> {
+        match filter {
+            (key, "=", value) if !key.is_empty() => Ok(PartitionFilter {
+                partition_key: key,
+                partition_value: PartitionValue::Equal(value),
+            }),
+            (key, "!=", value) if !key.is_empty() => Ok(PartitionFilter {
+                partition_key: key,
+                partition_value: PartitionValue::NotEqual(value),
+            }),
+            (_, _, _) => Err(DeltaTableError::InvalidPartitionFilter {
+                partition_filter: format!("{:?}", filter),
+            }),
+        }
+    }
+}
+
+impl<'a, T: std::fmt::Debug> TryFrom<(&'a str, &str, Vec<T>)> for PartitionFilter<'a, T> {
+    type Error = DeltaTableError;
+
+    fn try_from(filter: (&'a str, &str, Vec<T>)) -> Result<Self, DeltaTableError> {
+        match filter {
+            (key, "in", value) if !key.is_empty() => Ok(PartitionFilter {
+                partition_key: key,
+                partition_value: PartitionValue::In(value),
+            }),
+            (key, "not in", value) if !key.is_empty() => Ok(PartitionFilter {
+                partition_key: key,
+                partition_value: PartitionValue::NotIn(value),
+            }),
+            (_, _, _) => Err(DeltaTableError::InvalidPartitionFilter {
+                partition_filter: format!("{:?}", filter),
+            }),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct DeltaTablePartition<'a> {
+    pub key: &'a str,
+    pub value: &'a str,
+}
+
+impl<'a> DeltaTablePartition<'a> {
+    pub fn filter_accept_partition(&self, filter: &PartitionFilter<&'a str>) -> bool {
```

Comment:
nitpick, how about naming it as `match_filter`? The `partition` suffix here sounds redundant.

Reply:
👍